cfme/tests/openstack/cloud/test_volumes.py

```diff
@@ -0,0 +1,189 @@
+"""Tests for cloud provider's volumes"""
+
+import pytest
+from fauxfactory import gen_alphanumeric
+from cfme.cloud.instance import Instance
+from cfme.cloud.volume import (creation_form, device_input, get_volume_name,
+                               list_tbl, select_vm, select_volume, Volume)
+from cfme.fixtures import pytest_selenium
+from cfme.web_ui import InfoBlock, toolbar
+from cfme.web_ui.flash import assert_message_contain
+from cfme.web_ui.form_buttons import (attach_volume, detach_volume, FormButton,
+                                      submit_changes)
+from random import choice
+from utils import testgen, version
+from utils.appliance.implementations.ui import navigate_to
+from utils.wait import wait_for
+
+pytest_generate_tests = testgen.generate(testgen.provider_by_type,
+                                         ['openstack'], scope='module')
+
+
+PROD_VERSION = version.current_version().product_version()
+
+
+@pytest.mark.usefixtures("setup_provider_modscope")
+@pytest.mark.uncollectif(lambda: PROD_VERSION > '4.1')
```

Comment:
As I said on top, this will have the version check function call in it. I see this is on all the tests so what you might like to do is this:

```python
# on the module level

pytestmark = [pytest.mark.uncollectif(lambda: current_version() < ...)]
```

this will apply the mark on all the tests here.